### PR TITLE
CQ_DISPATCH_CMD_WRITE_PAGED support in test_dispatcher and passing tests

### PR DESCRIPTION
**Pull request by kmabeeTT**
Including the test_dispatcher/infra changes for testing CQ_DISPATCH_CMD_WRITE_PAGED that are (now passing more cases) that I excluded from previous merge. 

- Dispatcher test/infra updates (mostly new functions for paged write handling) for L1, DRAM and perf testing.
- A script for bunch of interesting testcases I've tried so far crossed with DRAM/L1, for easy verif, and comparisons (since these tests are not yet in CI, will work on that later and include a few of the more interesting cases here).

Thinking of keeping as 3 different commits, but can squash if we feel strongly.

WIP (Not included here) test_prefetcher.cpp changes.

### CODE REVIEW

**Comment by kmabeeTT** [comment]
Rebased on latest and resolved write -> write_linear conflicts in common.h (no biggie).

File: tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_dispatcher.cpp [hunk 1]
```diff
@@ -75,12 +90,20 @@ void init(int argc, char **argv) {
         log_info(LogTest, "    -b: dispatcher buffer size in blocks (default {})", DEFAULT_DISPATCH_BUFFER_SIZE_BLOCKS);
         log_info(LogTest, "  -pbs: prefetcher buffer size pages (default {})", DEFAULT_PREFETCHER_BUFFER_SIZE_PAGES);
         log_info(LogTest, " -ppbs: prefetcher page batch size (process pages in batches of N to reduce overhead) (default {})", DEFAULT_PREFETCHER_PAGE_BATCH_SIZE);
-        log_info(LogTest, "  -max: max transfer size bytes (default {})", MAX_XFER_SIZE_16B << 4);
-        log_info(LogTest, "  -min: min transfer size bytes (default {})", MIN_XFER_SIZE_16B << 4);
         log_info(LogTest, "    -f: prefetcher fire once, use to measure dispatcher perf w/ prefetcher out of the way (default disabled)");
-        log_info(LogTest, "    -d: wrap all commands in debug commands (default disabled)");
+        log_info(LogTest, "    -d: wrap all commands in debug commands and clear DRAM (default disabled)");
```
Comment (pgkeller) [inline tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_dispatcher.cpp:94]: not sure what "and clear DRAM" means

Comment (pgkeller) [reply tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_dispatcher.cpp:94]: oh, reading other code I see.  maybe "and clear DRAM to known state"

**Review by pgkeller — APPROVED** [review]

File: tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_dispatcher.cpp (hunk 1 above)
Comment (kmabeeTT) [reply tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_dispatcher.cpp:90]: thx, updated message.

**Pull request merged**